freqtrade/exchange/common.py

```diff
@@ -72,28 +76,34 @@ def calculate_backoff(retrycount, max_retries):
 def retrier_async(f):
     async def wrapper(*args, **kwargs):
         count = kwargs.pop('count', API_RETRY_COUNT)
+        self = args[0]  # Extract the exchange instance.
+        kucoin = self.name == "Kucoin"
         try:
             return await f(*args, **kwargs)
         except TemporaryError as ex:
-            logger.warning('%s() returned exception: "%s"', f.__name__, ex)
+            message = f'{f.__name__}() returned exception: "{ex}"'
+            (log_once_warning if kucoin else logger.warning)(message)
```

Comment:
That's a dangerous premise, which may result in important messages to be skipped (for example during downloading data - or other ratelimiting messages if rate-limit is set too low).

The **only place** i feel a cached logger is acceptable is within the IF block that captures the very specific 429000 kucoin error code.

All other exceptions (for example 429 without "429000") are valid rate-limiting exceptions - which i will expect to have logged.

Reply:
Reverted part of the code: removed all log_once cases except the 429000 kucoin error code one.